### PR TITLE
Refactor in order to support Maven plugin

### PR DESCRIPTION
**Pull request by patxibocos**
Key changes:

- App has been separated into two (and a future extra one) projects:
	- `core`: Contains the core business logic (generators)
	- `gradle-plugin`: It has just the Gradle plugin necessary classes
	- `maven-plugin`: To be built in the future
- The maven-plugin project will need to be a Maven project, so both gradle-plugin and maven-plugin have a dependency on core. The "problem" is that introducing a Gradle or Maven project in the other build system is not easy/smooth. So the proposed approach is to make the core and independent Gradle project that is built into a fat jar and deployed in a local Maven repository. So then both gradle and maven plugin projects will grab the core library from the local Maven.

### CODE REVIEW

**Comment by ralphpina** [comment]
Total nitpik: `Core` is a generic name, but maybe something like `generators` conveys more info when one is navigating the file system?

File: gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
 group = "com.patxi"
-version = poetimizelyVersion()
+version = poetimizelyVersion(project)
 
 plugins {
     `java-gradle-plugin`
     id("com.gradle.plugin-publish") version "0.11.0"
     jacoco
-    kotlin("jvm") version "1.3.70"
+    kotlin("jvm") version "1.3.72"
```
Comment (ralphpina) [inline gradle-plugin/build.gradle.kts:11]: unrelated change 😏

File: core/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    testImplementation("com.github.tschuchortdev:kotlin-compile-testing:1.2.7")
+    testImplementation("io.kotest:kotest-runner-junit5-jvm:4.0.5")
+    testImplementation("io.kotest:kotest-assertions-core-jvm:4.0.5")
+    testImplementation("io.mockk:mockk:1.9.3")
+    testImplementation(gradleTestKit())
+}
+
+buildscript {
+    apply(file("gradle/ktlint.gradle.kts"))
```
Comment (ralphpina) [inline core/build.gradle.kts:34]: this is great. I love the formatter.

File: gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (patxibocos-sh) [reply gradle-plugin/build.gradle.kts:11]: Haha yes 🙂

**Comment by patxibocos-sh** [comment]
> Total nitpik: `Core` is a generic name, but maybe something like `generators` conveys more info when one is navigating the file system?

You are right so I have pushed a new commit to rename it to `generator` 👍

File: gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (patxibocos) [reply gradle-plugin/build.gradle.kts:11]: Haha yes 🙂

**Comment by patxibocos** [comment]
> Total nitpik: `Core` is a generic name, but maybe something like `generators` conveys more info when one is navigating the file system?

You are right so I have pushed a new commit to rename it to `generator` 👍
